tracpro/groups/views.py

```diff
@@ -108,26 +108,31 @@ def post(self, request, *args, **kwargs):
             sent_parents = set(str(i) for i in data.values() if i is not None)
             if sent_regions != expected_ids:
                 msg = ("Data must map region id to parent id for each "
-                       "region in this org. ")
-                logger.warning("Hierarchy: " + msg + raw_data)
+                       "region in this org.")
+                logger.warning("{} Hierarchy: {} {}".format(org, msg, raw_data))
                 return self.error_response(400, msg)
             elif not sent_parents.issubset(expected_ids):
                 msg = ("Region parent must be a region from the same org, "
-                       "or null. ")
-                logger.warning("Hierarchy: " + msg + raw_data)
+                       "or null.")
+                logger.warning("{} Hierarchy: {} {}".format(org, msg, raw_data))
                 return self.error_response(400, msg)
 
-            # Re-set parent values for each region.
-            with Region.objects.delay_mptt_updates():
+            # Re-set parent values for each region, then rebuild the mptt tree.
+            with Region.objects.disable_mptt_updates():
```

Comment:
Have you tried wrapping this inside of a database transaction? The mptt docs suggest you do this.
https://django-mptt.github.io/django-mptt/mptt.managers.html#mptt.managers.TreeManager.disable_mptt_updates